src/buildstream/source.py

```diff
@@ -721,31 +727,64 @@ def translate_url(
+
         # Ensure that the download URL is also marked
         self.mark_download_url(url, primary=primary)
 
         if suffix:
             url = url + suffix
 
+        # Here we are called either:
+        #   * From a source fetcher with "alias_override" specified
+        #   * From a cloned source with self.__alias_override specified
+        #   * From the default source, with a possibly aliased source
+        #     * In this last case, mirrors are ignored
+        #
+        # In both relevant cases, we need to have the SourceMirror in context, and
+        # then we can delegate the translation to the SourceMirror while providing
+        # the original "url" to the SourceMirror method.
+        #
+        # Use the `self.__active_mirror` object
+
         # Alias overriding can happen explicitly (by command-line) or
         # implicitly (the Source being constructed with an __alias_override).
+        #
         if alias_override or self.__alias_override:
+
             url_alias, url_body = url.split(utils._ALIAS_SEPARATOR, 1)
-            if url_alias:
-                if alias_override:
-                    url = alias_override + url_body
-                else:
-                    # Implicit alias overrides may only be done for one
-                    # specific alias, so that sources that fetch from multiple
-                    # URLs and use different aliases default to only overriding
-                    # one alias, rather than getting confused.
-                    override_alias = self.__alias_override[0]  # type: ignore
-                    override_url = self.__alias_override[1]  # type: ignore
-                    if url_alias == override_alias:
-                        url = override_url + url_body
-            return url
+            project_alias_url = project.get_alias_url(url_alias, first_pass=self.__first_pass)
+
+            if self.__alias_override is not None:
+                override_alias = self.__alias_override[0]  # type: ignore
+                override_mirror = self.__alias_override[1]  # type: ignore
+
+                # Implicit alias overrides may only be done for one
+                # specific alias, so that sources that fetch from multiple
+                # URLs and use different aliases default to only overriding
+                # one alias, rather than getting confused.
+                #
+                if url_alias != override_alias:
+                    return url
+
+            elif alias_override is not None:
+                override_mirror = alias_override
+
+            assert override_mirror is not None, f"{alias_override} or {self.__alias_override}"
+
+            if isinstance(override_mirror, str):
+                return override_mirror + url_body
+            #
+            # Delegate the URL translation to the SourceMirror plugin
+            #
+            return override_mirror.translate_url(
+                project_name=project.name,
```

Comment:
The `project_name` parameter has been removed from `SourceMirror.translate_url()`.